packages/sync-actions/test/types-sync-fields.spec.js

```diff
@@ -0,0 +1,279 @@
+            label: {
+              en: 'EN field definition',
+            },
+          },
+        ],
+      })
+      updateActions = typesSync.buildActions(now, before)
+    })
+    test('should return `addFieldDefinition` action', () => {
+      expect(updateActions).toEqual(
+        now.fieldDefinitions.map(fieldDefinition => ({
+          action: 'addFieldDefinition',
+          fieldDefinition,
+        }))
+      )
+    })
+  })
+  describe('with fieldDefinitions removed', () => {
+    beforeEach(() => {
+      before = createTestType({
+        fieldDefinitions: [
+          {
+            type: { name: 'text' },
+            name: 'name-field-definition',
+            label: {
+              en: 'EN field definition',
+            },
+          },
+          {
+            type: { name: 'number' },
+            name: 'number-field-definition',
+            label: {
+              en: 'EN field definition',
+            },
+          },
+        ],
+      })
+      now = createTestType({
+        fieldDefinitions: [],
+      })
+      updateActions = typesSync.buildActions(now, before)
+    })
+    test('should return `removeFieldDefinition` actions', () => {
+      expect(updateActions).toEqual(
+        before.fieldDefinitions.map(fieldDefinition => ({
+          action: 'removeFieldDefinition',
+          fieldName: fieldDefinition.name,
+        }))
+      )
+    })
+  })
+  describe('with changing order of fieldDefinitions', () => {
+    beforeEach(() => {
+      before = createTestType({
+        fieldDefinitions: [
+          {
+            name: 'first',
+          },
+          {
+            name: 'second',
+          },
+        ],
+      })
+      now = createTestType({
+        fieldDefinitions: [
+          {
+            name: 'second',
+          },
+          {
+            name: 'first',
+          },
+        ],
+      })
+      updateActions = typesSync.buildActions(now, before)
+    })
+    test('should return `changeFieldDefinitionOrder` action', () => {
+      expect(updateActions).toEqual([
+        {
+          action: 'changeFieldDefinitionOrder',
+          fieldNames: [{ name: 'second' }, { name: 'first' }],
+        },
+      ])
+    })
+  })
+  describe('with fieldDefinitions replaced', () => {
+    beforeEach(() => {
+      before = createTestType({
+        fieldDefinitions: [
+          {
+            type: { name: 'number' },
+            name: 'number-field-definition',
+            label: {
+              en: 'number-en',
+            },
+          },
+        ],
+      })
+      now = createTestType({
+        fieldDefinitions: [
+          {
+            type: { name: 'text' },
+            name: 'text-field-definition',
+            label: {
+              en: 'text-en',
+            },
+          },
+        ],
+      })
+      updateActions = typesSync.buildActions(now, before)
+    })
+    test('should return `removeFieldDefinition` and `addFieldDefinition` actions', () => {
```

Comment:
I would like to see a test with the following data:

```js
previous = [
  {
    // no change
  },
  {
    // to be removed
  },
  {
    // will change to 'z'
    name: 'x'
  },
]
next = [
  {
    // no change
  },
  {
    // changed from 'x'
    name: 'z'
  },
]
````

Reply:
There is no update action for changing the `name`, only the `label`([ref](https://docs.commercetools.com/http-api-projects-types.html#change-fielddefinition-label)). 

Anyways, added that test here https://github.com/commercetools/nodejs/pull/757/commits/6b3da242b3d8d38f5ea7596781f3c6ea4f230bcb

Reply:
@adnasa added here https://github.com/commercetools/nodejs/pull/757/commits/f369ff0feb5431701dc4ddaea61ecdd47068f934